### PR TITLE
Add AGB live woody density dataset

### PR DESCRIPTION
**Pull request by nilsleh**
This PR adds the [AGB density dataset](https://data.globalforestwatch.org/datasets/gfw::aboveground-live-woody-biomass-density/about). It contains global AGB density values (megagrams biomass/hectare) at ~30m resolution.

Dataset features:
* Masks with per pixel live woody biomass density estimates in megagrams biomass per hectare at ~30m resolution (~40,000x40,0000 px)

Dataset format:
* geojson file that contains download links to tif files
* single-channel geotiffs with the pixel values representing biomass density

Example:
woodDensity

### CODE REVIEW

File: docs/api/datasets.rst [hunk 1]
```diff
@@ -12,6 +12,11 @@ Geospatial Datasets
 
 :class:`GeoDataset` is designed for datasets that contain geospatial information, like latitude, longitude, coordinate system, and projection. Datasets containing this kind of information can be combined using :class:`IntersectionDataset` and :class:`UnionDataset`.
 
+AbovegroundLiveWoodyBiomassDensity
```
Comment (calebrob6) [inline docs/api/datasets.rst:15]:
```suggestion
Aboveground Live Woody Biomass Density
```

File: torchgeo/datasets/agb_live_woody_density.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+import matplotlib.pyplot as plt
+from rasterio.crs import CRS
+from torch import Tensor
+
+from .geo import RasterDataset
+from .utils import download_url
+
+
+class AbovegroundLiveWoodyBiomassDensity(RasterDataset):
+    """AbovegroundLiveWoodyBiomassDensity dataset.
```
Comment (calebrob6) [inline torchgeo/datasets/agb_live_woody_density.py:20]:
```suggestion
    """Aboveground Live Woody Biomass Density dataset.
```

File: torchgeo/datasets/agb_live_woody_density.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+    The `AbovegroundLiveWoodyBiomassDensity dataset
```
Comment (calebrob6) [inline torchgeo/datasets/agb_live_woody_density.py:22]:
```suggestion
    The `Aboveground Live Woody Biomass Density dataset
```

**Review by calebrob6 — APPROVED** [review]

File: docs/api/datasets.rst [hunk 4]
```diff
@@ -12,6 +12,11 @@ Geospatial Datasets
 
 :class:`GeoDataset` is designed for datasets that contain geospatial information, like latitude, longitude, coordinate system, and projection. Datasets containing this kind of information can be combined using :class:`IntersectionDataset` and :class:`UnionDataset`.
 
+Aboveground Live Woody Biomass Density
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (adamjstewart) [inline docs/api/datasets.rst:16]:
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: torchgeo/datasets/agb_live_woody_density.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    url = (
+        "https://opendata.arcgis.com/api/v3/datasets/3e8736c8866b458687"
+        "e00d40c9f00bce_0/downloads/data?format=geojson&spatialRefId=4326"
+    )
+
+    base_filename = "Aboveground_Live_Woody_Biomass_Density.geojson"
+
+    filename_glob = "*N_*E.*"
+    filename_regex = (
+        r"""^(?P<latitude>[0-9][0-9][A-Z])_(?P<longitude>[0-9][0-9][0-9][A-Z])*"""
```
Comment (adamjstewart) [inline torchgeo/datasets/agb_live_woody_density.py:56]: Can you split this into multiple lines?

File: torchgeo/datasets/agb_live_woody_density.py [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+        for item in content["features"]:
+            download_url(
+                item["properties"]["download"],
+                self.root,
+                item["properties"]["tile_id"] + ".tif",
+            )
+
+    def plot(  # type: ignore[override]
+        self,
+        sample: Dict[str, Tensor],
```
Comment (adamjstewart) [inline torchgeo/datasets/agb_live_woody_density.py:137]:
```suggestion
        sample: Dict[str, Any],
```
See my comment on your other PR

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: torchgeo/datasets/agb_live_woody_density.py [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+
+        # Download the dataset
+        self._download()
+        self._extract()
+
+    def _download(self) -> None:
+        """Download the dataset."""
+        download_url(self.url, self.root, self.base_filename)
+
+    def _extract(self) -> None:
```
Comment (adamjstewart) [inline torchgeo/datasets/agb_live_woody_density.py:124]: Needs a docstring

File: torchgeo/datasets/agb_live_woody_density.py [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+        with open(os.path.join(self.root, self.base_filename), "r") as f:
+            content = json.load(f)
+
+        for item in content["features"]:
+            download_url(
```
Comment (adamjstewart) [inline torchgeo/datasets/agb_live_woody_density.py:129]: I think this should all go in `_download`, if there aren't any compressed files then you don't need a separate `_extract` method

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**